### PR TITLE
Saas 12.5 related binary rco seb

### PR DESCRIPTION
**Pull request by seb-odoo**
rebase-ff or do we write a merge message here and a proper title?

### CODE REVIEW

File: odoo/addons/base/models/ir_attachment.py [hunk 1]
```diff
@@ -500,12 +501,25 @@ def unlink(self):
...
+                data = values.pop('datas')
+                bin_data = base64.b64decode(data) if data else b''
+                values['file_size'] = len(bin_data)
+                values['checksum'] = self._compute_checksum(bin_data)
+                values['index_content'] = self._index(bin_data, values['mimetype'])
+                values['store_fname'] = False
+                values['db_datas'] = data
+                if data and location != 'db':
+                    values['store_fname'] = self._file_write(data, values['checksum'])
+                    values['db_datas'] = False
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_attachment.py]: factor this in a method, also used by inverse_datas

File: odoo/fields.py [hunk 2]
```diff
@@ -1910,27 +1913,6 @@ def read(self, records):
...
-                    'res_model': self.model_name,
-                    'res_field': self.name,
-                    'res_id': record.id,
-                    'type': 'binary',
-                    'datas': value,
-                }
-                for record, value in record_values
-                if value
-            ])
-
```
Comment (rco-odoo) [inline odoo/fields.py]: please leave this

File: odoo/fields.py [hunk 3]
```diff
@@ -1982,16 +1970,6 @@ class Image(Binary):
...
-    def create(self, record_values):
-        new_record_values = []
-        for record, value in record_values:
-            # strange behavior when setting related image field, when `self`
-            # does not resize the same way as its related field
-            new_value = self._image_process(value)
-            new_record_values.append((record, new_value))
-            record.env.cache.update(record, self, [value if self.related else new_value] * len(record))
-        super(Image, self).create(new_record_values)
-
```
Comment (rco-odoo) [inline odoo/fields.py]: please leave this one, too

File: odoo/models.py [hunk 4]
```diff
@@ -3739,7 +3741,8 @@ def _create(self, data_list):
                     cachetoclear.append((record, field))
                 else:
                     cache_value = field.convert_to_cache(value, record)
-                    self.env.cache.set(record, field, cache_value)
+                    if field.column_type:
+                        self.env.cache.set(record, field, cache_value)
```
Comment (rco-odoo) [inline odoo/models.py]: revert this part

File: odoo/addons/base/models/ir_attachment.py (hunk 1 above)
Comment (seb-odoo) [reply odoo/addons/base/models/ir_attachment.py]: Done

File: odoo/fields.py (hunk 2 above)
Comment (seb-odoo) [reply odoo/fields.py]: Reverted

File: odoo/fields.py (hunk 3 above)
Comment (seb-odoo) [reply odoo/fields.py]: Also reverted

File: odoo/models.py (hunk 4 above)
Comment (seb-odoo) [reply odoo/models.py]: And also reverted

**Comment by seb-odoo** [comment]
Runbot green here + enterprise

**Comment by seb-odoo** [comment]
@robodoo retry

**Comment by seb-odoo** [comment]
Rebased. Can't reproduce issue locally.

**Comment by rco-odoo** [comment]
@robodoo r+

**Comment by rco-odoo** [comment]
@seb-odoo the issue should be fixed by #37341.

**Comment by seb-odoo** [comment]
Thank you for the notice, I was about to investigate further.

Apparently the failing test has been disabled from staging branches.

@robodoo retry

**Pull request closed without merging**